packages/lib-classifier/src/store/SubjectStore/SubjectStore.spec.js

```diff
@@ -370,6 +371,34 @@ describe('Model > SubjectStore', function () {
     })
   })
 
+  describe('imange and text subjects', function () {
+    let subjects
+    const imageAndTextSubjects = Factory.buildList(
+      'subject',
+      10,
+      {
+        content: 'This is test subject content',
+        contentLoadingState: asyncStates.success,
+        locations: [
+          { 'image/png': 'https://foo.bar/example.png' },
+          { 'text/plain': 'https://foo.bar/example.txt' }
+        ]
+      })
+
+    before(async function () {
+      subjects = await mockSubjectStore(imageAndTextSubjects)
+    })
+
+    it('should be valid subjects', function () {
+      const expectedSubject = ImageAndTextSubject.create(imageAndTextSubjects[1])
+      expect(subjects.resources.get(expectedSubject.id)).to.deep.equal(expectedSubject)
+    })
+
+    it('should be of the correct subject type', function () {
+      expect(getType(subjects.active).name).to.equal('ImageAndTextSubject')
```

Comment:
This is the check that I think you can also use to determine which type of viewer to return for a subject.